iconservice/deploy/icon_score_deploy_engine.py

```diff
@@ -244,121 +176,132 @@ def _on_deploy(self,
         content_type: str = data.get('contentType')
-        content: bytes = data.get('content')
+        # content is a string on tbears mode, otherwise bytes
+        content = data.get('content')
         params: dict = data.get('params', {})
 
-        deploy_info = IconScoreContextUtil.get_deploy_info(context, tx_params.score_address)
+        deploy_info: 'IconScoreDeployInfo' =\
+            self.icon_deploy_storage.get_deploy_info(context, tx_params.score_address)
         if deploy_info is None:
-            next_tx_hash = None
+            next_tx_hash: bytes = ZERO_TX_HASH
         else:
-            next_tx_hash = deploy_info.next_tx_hash
-        if next_tx_hash is None:
-            next_tx_hash = bytes(DEFAULT_BYTE_SIZE)
+            next_tx_hash: bytes = deploy_info.next_tx_hash
 
         if content_type == 'application/tbears':
-            score_root_path = IconScoreContextUtil.get_score_root_path(context)
-            target_path = path.join(score_root_path,
-                                    score_address.to_bytes().hex())
-            makedirs(target_path, exist_ok=True)
-            converted_tx_hash: str = f'0x{bytes.hex(next_tx_hash)}'
-            target_path = path.join(target_path, converted_tx_hash)
-            try:
-                symlink(content, target_path, target_is_directory=True)
-            except FileExistsError:
-                pass
+            write_score_to_score_deploy_path: callable =\
+                self._write_score_to_score_deploy_path_on_tbears_mode
         else:
-            revision = IconScoreContextUtil.get_revision(context)
-
-            if revision >= REVISION_4:
-                install_path = DirectoryNameChanger.get_score_path_by_address_and_tx_hash(
-                    self._icon_score_deployer.score_root_path, score_address, next_tx_hash)
-                DirectoryNameChanger.rename_directory(install_path)
-                self._icon_score_deployer.deploy(
-                    address=score_address,
-                    data=content,
-                    tx_hash=next_tx_hash)
-            elif revision >= REVISION_2:
-                self._icon_score_deployer.deploy(
-                    address=score_address,
-                    data=content,
-                    tx_hash=next_tx_hash)
-            else:
-                self._icon_score_deployer.deploy_legacy(
-                    address=score_address,
-                    data=content,
-                    tx_hash=next_tx_hash)
+            write_score_to_score_deploy_path: callable =\
+                self._write_score_to_score_deploy_path
+        write_score_to_score_deploy_path(context, score_address, next_tx_hash, content)
 
         backup_msg = context.msg
         backup_tx = context.tx
+        new_score_mapper: 'IconScoreMapper' = context.new_icon_score_mapper
 
         try:
             if IconScoreContextUtil.is_service_flag_on(context, IconServiceFlag.SCORE_PACKAGE_VALIDATOR):
```

Comment:
make function one try_score_package_validate
insert this logic to try_score_package_validate

Reply:
Done